src/core/qgsvectorfilewriter.cpp

```diff
@@ -1551,6 +1565,10 @@ bool QgsVectorFileWriter::driverMetadata( const QString& driverName, QgsVectorFi
   return false;
 }
 
+OGRwkbGeometryType QgsVectorFileWriter::ogrTypeFromWkbType( QgsWKBTypes::Type type )
+{
+  return static_cast<OGRwkbGeometryType>( type );
```

Comment:
I'm not 100% sure, but I think it does need to convert. See http://www.gdal.org/ogr__core_8h.html#a800236a0d460ef66e687b7b65610f12a for the supported types.